test/benchmark.py

```diff
@@ -0,0 +1,327 @@
+        self._create_table()
+
+        self._one_row = (
+            1,
+            "Pete",
+            "Allison",
+            datetime.datetime(1998, 10, 6).strftime("%Y-%m-%d"),
+            b"123",
+        )
+
+    def _cleanup(self):
+        """Drop the test table."""
+        conn = spanner_dbapi.connect("sqlalchemy-dialect-test", "compliance-test")
+        conn.database.update_ddl(["DROP TABLE Singers"])
+        conn.close()
+
+    def _create_table(self):
+        """Create a table for performace testing."""
+        conn = spanner_dbapi.connect("sqlalchemy-dialect-test", "compliance-test")
+        conn.database.update_ddl(
+            [
+                """
+CREATE TABLE Singers (
+    id INT64,
+    first_name STRING(1024),
+    last_name STRING(1024),
+    birth_date DATE,
+    picture BYTES(1024),
+) PRIMARY KEY (id)
+        """
+            ]
+        ).result(120)
+
+        conn.close()
+
+    def run(self):
+        """Execute every test case."""
+        measures = {}
+        for method in (
+            self.insert_one_row_with_fetch_after,
+            self.read_one_row,
+            self.insert_many_rows,
+            self.select_many_rows,
+            self.insert_many_rows_with_mutations,
+        ):
+            method(measures)
+
+        self._cleanup()
+        return measures
+
+
+class SpannerBenchmarkTest(BenchmarkTestBase):
+    """The original Spanner performace testing class."""
+
+    def __init__(self):
+        super().__init__()
+        self._client = Client()
+        self._instance = self._client.instance("sqlalchemy-dialect-test")
+        self._database = self._instance.database("compliance-test2")
+
+        self._many_rows = []
+        self._many_rows2 = []
+        birth_date = datetime.datetime(1998, 10, 6).strftime("%Y-%m-%d")
+        for i in range(99):
+            num = round(random.random() * 1000000)
+            self._many_rows.append((num, "Pete", "Allison", birth_date, b"123"))
+            num2 = round(random.random() * 1000000)
+            self._many_rows2.append((num2, "Pete", "Allison", birth_date, b"123"))
+
+        # initiate a session
+        with self._database.snapshot():
+            pass
+
+    @measure_execution_time
+    def insert_one_row_with_fetch_after(self):
+        self._database.run_in_transaction(insert_one_row, self._one_row)
+
+    @measure_execution_time
+    def insert_many_rows(self):
+        self._database.run_in_transaction(insert_many_rows, self._many_rows)
+
+    @measure_execution_time
+    def insert_many_rows_with_mutations(self):
+        with self._database.batch() as batch:
+            batch.insert(
+                table="Singers",
+                columns=("id", "first_name", "last_name", "birth_date", "picture"),
+                values=self._many_rows2,
+            )
+
+    @measure_execution_time
+    def read_one_row(self):
+        with self._database.snapshot() as snapshot:
+            keyset = KeySet(keys=([1],))
```

Comment:
In the interest of being equivalent, this should probably be:
```suggestion
            keyset = KeySet(all_=True)
```
The SQLAlchemy SQL is executing `SELECT * FROM Singers` while I think this is currently executing the equivalent of `SELECT * FROM Singers WHERE id = 1`